woocommerce/wc-admin
# Store all order statuses

Fixes #1251 

This PR:
* Stores all order statuses
* Adds an array of excluded statuses
* Filters all products, coupons, categories, taxes, variations, and orders by the default (non-excluded) statuses if no status query is provided

### Detailed test instructions:

1.  Make sure that you have regenerated lookup tables and that the `status` column is filled in `wc_order_stats`.
2.  Create requests to each of the endpoints.
3. Check that stats returned included stats from all orders excluding the default excluded statuses (`refunded`, `pending`, `failed`, `cancelled`).
4.  Make a request to the `/wp-json/wc/v3/reports/orders/stats` endpoint and check that `status_is` and `status_is_not` query params override default excluded statuses (e.g., you can still query by `failed` order statuses).
5.  Check that all phpunit tests are passing.
